StockCount/View/EditViewController.swift

```diff
@@ -24,26 +26,71 @@ class EditViewController: CustomNavBarViewController, UITextViewDelegate
     var keyboardToolBar : KeyboardToolBar?
     private let appleKeyboardIdentifier = ["en_US@hw=Automatic;sw=QWERTY", "en_US@sw=QWERTY;hw=Automatic"]
     
+    let volumeControl = MPVolumeView(frame: CGRect(x: 0, y: 0, width: 120, height: 120))
+    var isForceVolumeChange = false
+    
     override func viewDidLoad() {
         super.viewDidLoad()
         
         updateTopButtons(isDoneVisible: false)
         
         //Keyboard Notification
-        NotificationCenter.default.addObserver(self, selector: #selector(self.keyboardNotification(notification:)), name: NSNotification.Name.UIKeyboardWillChangeFrame, object: nil)
+        NotificationCenter.default.addObserver(self, selector: #selector(self.keyboardNotification(notification:)), name: UIResponder.keyboardWillShowNotification, object: nil)
         
         keyboardToolBar = KeyboardToolBar(frame: CGRect(x: 0, y: 0, width: self.view.frame.size.width, height: 50))
         keyboardToolBar?.delegate = self
         txtView.inputAccessoryView = keyboardToolBar
         
         editController.readFile(fileName)
+        
+        view.addSubview(volumeControl)
+    }
+    override func viewDidLayoutSubviews() {
+        volumeControl.frame = CGRect(x: -120, y: -120, width: 100, height: 100)
     }
     
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+    }
     override func viewDidAppear(_ animated: Bool) {
         super.viewDidAppear(animated)
         txtView.becomeFirstResponder()
+        
+        AVAudioSession.sharedInstance().addObserver(self, forKeyPath: "outputVolume", options: .new, context: nil)
+        
+        setVolume(0.5)
     }
+    
+    override func viewDidDisappear(_ animated: Bool) {
+        super.viewDidDisappear(animated)
         
+        AVAudioSession.sharedInstance().removeObserver(self, forKeyPath: "outputVolume")
+    }
+    
+    override func observeValue(forKeyPath keyPath: String?, of object: Any?, change: [NSKeyValueChangeKey : Any]?, context: UnsafeMutableRawPointer?) {
+        switch keyPath {
+        case "outputVolume":
+            if !isForceVolumeChange {
+                didTriggerScan()
+            } else {
+                isForceVolumeChange = false
+            }
+            if change != nil {
+                let volume = change![NSKeyValueChangeKey.newKey] as? Float
+                if volume == 1 || volume == 0 {
```

Comment:
What are those magic constants?? Please use a define to understand what is `1` ans `0`.

Reply:
I think that's because app stops getting volume change notification when the volume reaches to 0 or 1.